spec/query_spec.cr

```diff
@@ -156,6 +156,17 @@ describe LuckyRecord::Query do
 
       query.count.should eq 1
     end
+
+    it "raises when used with offset or limit" do
+      msg = "Counting with limit or offset is not supported yet. Try calling `results.size` to count in Crystal instead."
+      expect_raises Exception, msg do
```

Comment:
I'm fine with the `UnsupportedSQLError` or this, but either way you can make these expectations simpler by using a subset of the message: `expect_raises Exception, "not support yet"`

This works because the `expect_raises` helper uses String#includes? instead of equality: https://github.com/crystal-lang/crystal/blob/204bfd0555921f3aadbda289993cca5323ebaf95/src/spec/expectations.cr#L272

This makes the tests a lot less brittle. I hope that little tidbit is helpful for you to know :)

Reply:
That's definitely a useful feature and I try to use it as much as possible. I guess it's enough for most cases, and the main point of using an `Exception` class is to be able to rescue it explicitly, which is good to know